### PR TITLE
add disambig f1 score

### PR DESCRIPTION
**Pull request by bugtig6351**
https://github.com/gomate-community/rageval/issues/29#issuecomment-1976102556

### CODE REVIEW

File: rageval/metrics/_answer_disambig_f1.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    'AnswerCorrectness'
+    >>> s, ds = metric.compute(dataset, batch_size=1)
+    >>> assert 0 <= s <= 1
+    >>> type(ds)
+    <class 'datasets.arrow_dataset.Dataset'>
+"""
+
+_CITATION = """\
+@misc{jiang2023active,
+      title={Active Retrieval Augmented Generation},
```
Comment (faneshion) [inline rageval/metrics/_answer_disambig_f1.py:61]: The citation should be this paper (i.e., ASQA: Factoid Questions Meet Long-Form Answers).

File: rageval/metrics/_answer_disambig_f1.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    def _validate_data(self, dataset: datasets.Dataset) -> bool:
+        """Validate the of the input dataset."""
+        super()._validate_data(dataset)
+        if not all(isinstance(answer, str) for answer in dataset["answers"]):
+            raise ValueError("The type of answers should be a string.")
+        if not all(isinstance(a, List) or not all(isinstance(item, str) for item in a) for a in dataset["gt_answers"]):
+            raise ValueError("The type of gt_answers should be a list of strings.")
+
+    def _f1_score(self, pred: str, ref: str) -> float:
+        """Compute the f1 score between pred and ref."""
```
Comment (faneshion) [inline rageval/metrics/_answer_disambig_f1.py:143]: The original version of disambig-F1 in the paper is built on Roberta trained on SQUADV2, and you replace it with exact match?

File: requirements.txt [hunk 3]
```diff
@@ -16,3 +16,5 @@ transformers == 4.37.2
 torch == 2.2.0
 pandas == 2.0.0
 nltk == 3.8.1
+spacy == 3.7.4
+en_core_web_sm @ https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.7.1/en_core_web_sm-3.7.1-py3-none-any.whl
```
Comment (faneshion) [inline requirements.txt:20]: Is this package a model? what size is it?

Comment (bugtig6351) [reply requirements.txt:20]: Yes, the size of model is 12.2MB. More details: https://github.com/explosion/spacy-models/releases

File: rageval/metrics/_answer_disambig_f1.py (hunk 2 above)
Comment (bugtig6351) [reply rageval/metrics/_answer_disambig_f1.py:143]: The original metric was presented in [ASQA paper](https://aclanthology.org/2022.emnlp-main.566/), and implemented through [this code](https://github.com/google-research/language/blob/master/language/asqa/scoring.py#L273). And we adopted an [alternative implementation](https://github.com/jzbjyb/FLARE/blob/ec4b06b502b5ab54f3f9236b0112a5d28482e7bb/src/datasets.py#L29) from the paper [Active Retrieval Augmented Generation](https://arxiv.org/abs/2305.06983).

**Review by faneshion — APPROVED** [review]

**Pull request merged**